Solutions/CohesitySecurity/Data Connectors/Helios2Sentinel/IncidentProducer/IncidentProducer.cs

```diff
@@ -0,0 +1,263 @@
+#else
+            [TimerTrigger("* */5 * * * *")]TimerInfo myTimer,
+#endif
+            [Queue("%CohesityQueueName%"), StorageAccount("AzureWebJobsStorage")] ICollector<string> outputQueueItem,
+            ILogger log)
+        {
+            log.LogInformation($"C# Timer trigger function executed at: {DateTime.Now}");
+            long startDateUsecs = 0;
+
+            try
+            {
+                var db = Connection.GetDatabase();
+                string apiKey = GetSecret("ApiKey");
+                string blobKey = GetSecret("workspace") + "\\" + apiKey;
+
+                try
+                {
+                    startDateUsecs = long.Parse(GetData(blobKey));
+                }
+                catch  (Exception ex)
+                {
+                    startDateUsecs = GetPreviousUnixTime(log);
+                    log.LogError("Exception --> 2 " + ex.Message);
+                }
+
+                if (startDateUsecs == 0)
+                {
+                    startDateUsecs = GetPreviousUnixTime(log);
+                }
+
+                log.LogInformation ("startDateUsecs --> " + startDateUsecs);
+
+                long endDateUsecs = GetCurrentUnixTime();
+                log.LogInformation ("endDateUsecs --> " + endDateUsecs.ToString());
+
+                string requestUriString = $"https://helios.cohesity.com/mcm/alerts?alertCategoryList=kSecurity&alertStateList=kOpen&startDateUsecs={startDateUsecs}&endDateUsecs={endDateUsecs}";
+                log.LogInformation("requestUriString --> " + requestUriString);
+                using HttpClient client = new ();
+                client.DefaultRequestHeaders.Accept.Clear();
+                client.DefaultRequestHeaders.Add("apiKey", GetSecret("apiKey"));
+                await using Stream stream = await client.GetStreamAsync(requestUriString);
+                StreamReader reader = new StreamReader(stream);
+                dynamic alerts = JsonConvert.DeserializeObject(reader.ReadToEnd());
+
+                var tasks = new List<Task>();
+
+                foreach (var alert in alerts)
+                {
+                    tasks.Add(Task.Run( () =>
+                    {
+                        ParseAlertToQueue(outputQueueItem, alert, log);
+                    }));
+                }
+                Task t = Task.WhenAll(tasks);
+                try
+                {
+                    t.Wait();
+                }
+                catch {}
+
+                if (t.Status == TaskStatus.RanToCompletion)
+                {
+                    WriteData(blobKey, endDateUsecs.ToString(), log);
+                }
+                log.LogInformation("new startDateUsecs --> " + endDateUsecs.ToString());
+            }
+            catch  (Exception ex)
+            {
+                log.LogError("Exception --> 3 " + ex.Message);
+            }
+        }
+
+        private static string GetSecret(string secretName)
```

Comment:
You have exactly the same function in IncidentConsumer. I'd create a library project, put this function there and add the dependency to IncidentConsumer and IncidentProducer.
Alternatively, you can extract this function to a separate class in a separate file and link it to both projects (see explanations at [stackoverflow.com/questions/1116465/how-do-you-share-code-between-projects-solutions-in-visual-studio](https://stackoverflow.com/questions/1116465/how-do-you-share-code-between-projects-solutions-in-visual-studio))